core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeGrtrThanEquaToFilterExecuterImpl.java

```diff
@@ -149,13 +154,17 @@ private boolean isScanRequired(byte[] blockMaxValue, byte[][] filterValues) {
     return isScanRequired;
   }
 
-  private boolean isScanRequired(byte[] maxValue, Object[] filterValue,
+  private boolean isScanRequired(byte[] maxValue, boolean nullValue, Object[] filterValue,
       DataType dataType) {
     Object value = DataTypeUtil.getMeasureObjectFromDataType(maxValue, dataType);
     for (int i = 0; i < filterValue.length; i++) {
       // TODO handle min and max for null values.
       if (filterValue[i] == null) {
-        return true;
+        if (nullValue == true) {
+          return true;
+        } else {
+          return false;
+        }
```

Comment:
you no need use if condition, just return nullValue

Reply:
Done